kafka-client/src/main/java/dev/responsive/kafka/internal/stores/AbstractResponsiveStore.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.stores;
+
+import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.asInternalProcessorContext;
+
+import dev.responsive.kafka.internal.db.CassandraClient;
+import dev.responsive.kafka.internal.utils.TableName;
+import java.util.Collection;
+import java.util.function.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.StateStoreContext;
+import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+import org.apache.kafka.streams.query.Position;
+
+public abstract class AbstractResponsiveStore implements StateStore {
```

Comment:
For reviewers: the introduction of this abstract class is the main change here. Tried to pull out all/only the common logic between the stores. There might be some further adjustments to how things are broken up in `init` when I rebase the standby-task fix on top of this, but I wanted to make sure this PR stands on its own in case that fix takes any longer